### DISCUSSION TITLE
Network request interception on chrome/firefox does not show changes

### DISCUSSION THREAD

**Discussion by ghost** [mitmproxy/mitmproxy#6499] [General]
I've spent hours trying to find an answer to this thinking the code is not working. Sample code below. Its just a simple rewrite of request body but when I checked chrome dev tools network tab or firefox, I can't see the changes being made my mitmproxy. Nothing has changed to the request body although the code is working. Is that normal to not see the changes in the network request for chrome/firefox?
```
from mitmproxy import http

def request(flow: http.HTTPFlow) -> None:
    if flow.request.urlencoded_form:
        flow.request.urlencoded_form["mitmproxy"] = "rocks"
```

**Comment by Prinzhorn** [comment]
> Is that normal to not see the changes in the network request for chrome/firefox?

Yes

browser -> mitmproxy -> server

if you change the request on it's way to the server in mitmproxy, how would the browser know?

**Reply by ghost** [reply]
That make sense. I thought the changes will also show to the browsers dev tools . Thank you!
